### PR TITLE
Bump go version to 1.22

### PR DESCRIPTION
**Pull request by urvisavla**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Bump go version to 1.22 

### Why

https://github.com/stellar/go/issues/5230

### Known limitations

N/A

### CODE REVIEW

**Comment by sreuland** [comment]
@urvisavla , can you target this pr at new [2.28.4 rel branch](https://github.com/stellar/go/tree/release-horizon-v2.28.4)

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-22.04]
-        go: ["1.21"]
+        go: ["1.22.1"]
```
Comment (sreuland) [inline .github/workflows/go.yml:21]: is there a reason to pin the minor ver`1.22.1` or can it be `1.22` ?

File: services/horizon/docker/verify-range/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,6 @@
 FROM ubuntu:22.04
 
+ARG GO_VERSION
```
Comment (sreuland) [inline services/horizon/docker/verify-range/Dockerfile:3]: not seeing a reference to this ARG, can it be removed?

File: .github/workflows/go.yml (hunk 1 above)
Comment (urvisavla) [reply .github/workflows/go.yml:21]: Like I mentioned in the slack I'm not entirely sure why that was required but this the "check" action was failing without it.

**Comment by urvisavla** [comment]
> @urvisavla , can you target this pr at new [2.28.4 rel branch](https://github.com/stellar/go/tree/release-horizon-v2.28.4), thanks!

Sure if we're planning to get  2.28.4 out in a day or two; otherwise it might be better to include this in 2.28.3 considering it's related to a CVE?

**Pull request merged**